### ISSUE TITLE
no-fallthrough reports false positive when commenting cases

### ISSUE THREAD

**Issue by steveluscher** [eslint/eslint#13339]
Consider the following code:

```
var value = 'ONE';
switch (value) {
  case 'ONE':
  default:
    1 == 1;
    break;
  case 'TWO':
    1 == 1;
    break; 
}
```

Lint is fine with that, until such time as you try to add a comment admidst the cases.

```
var value = 'ONE';
switch (value) {
  case 'ONE':
  // A helpful comment.
  default:
    1 == 1;
    break;
  case 'TWO':
    1 == 1;
    break; 
}
```

…at which point the `no-fallthrough` error is raised.

A comment here doesn't affect the runtime behavior, right, so to error here is a false positive?

Repro [link](https://eslint.org/demo#eyJ0ZXh0IjoidmFyIHZhbHVlID0gJ09ORSc7XG5zd2l0Y2ggKHZhbHVlKSB7XG4gIGNhc2UgJ09ORSc6XG4gIC8vIFJlbW92ZSB0aGlzIGNvbW1lbnQgdG8gc2F0aXNmeSBsaW50LlxuICBkZWZhdWx0OlxuXHQxID09IDE7XG4gICAgYnJlYWs7XG4gIGNhc2UgJ1RXTyc6XG5cdDEgPT0gMTtcbiAgICBicmVhazsgXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1lbHNlLWlmIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbXBvcnQtYXNzaWduIjoyLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOjIsIm5vLWludmFsaWQtcmVnZXhwIjoyLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjoyLCJuby1yZWRlY2xhcmUiOjIsIm5vLXJlZ2V4LXNwYWNlcyI6Miwibm8tc2VsZi1hc3NpZ24iOjIsIm5vLXNldHRlci1yZXR1cm4iOjIsIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZWxlc3MtY2F0Y2giOjIsIm5vLXVzZWxlc3MtZXNjYXBlIjoyLCJuby13aXRoIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJ1c2UtaXNuYW4iOjIsInZhbGlkLXR5cGVvZiI6Mn0sImVudiI6e319fQ==).

**Comment by mdjermanovic** [comment]
Hi @steveluscher, thanks for the issue!

It seems that this works as intended, please see the discussion in #5799.

There is also a [test case](https://github.com/eslint/eslint/blob/3a5fbb3d634be688615950c0a5fa8aead6ff08b5/tests/lib/rules/no-fallthrough.js#L137) similar to this example.

The following is allowed:

```js
var value = 'ONE';
switch (value) {
  case 'ONE': // A helpful comment.  
  default:
    1 == 1;
    break;
  case 'TWO':
    1 == 1;
    break; 
}
```

Also, [commentPattern](https://eslint.org/docs/rules/no-fallthrough#commentpattern) is configurable, which might be helpful for comments on separate lines.
